core/dev/spi.h

```diff
@@ -54,39 +54,46 @@
 
 extern unsigned char spi_busy;
 
+/* Deprecated */
 void spi_init(void);
+/*---------------------------------------------------------------------------*/
+/* New SPI API supporting mutliple instances.
+   See /cpu/cc2538/spi-arch.h + cpu/cc2538/dev/spi.c for an example on
+   how this new API could be implemented. */
+void spix_init(uint8_t instance);
```

Comment:
Good point. Only the CPU-specific declarations and definitions should be in `spi-arch.h`. The official SPI API should be in `spi.h`. The `spix` API is only a temporary thing before we come up with a better SPI API, so maybe it could be kept in `spi-arch.h`.